### DISCUSSION TITLE
Please help me. Failed to use ssh key to pull code repository

### DISCUSSION THREAD

**Discussion by dizuncainiao** [community/community#88410] [Repositories] [answered]
### Select Topic Area

Question

### Body

Report this error：
`
ssh: connect to host github.com port 22: Connection timed out Could not read from remote repository.  Please make sure you have the correct access rights and the repository exists.
`
I reconfigured ssh key but it doesn't work. I'm sure it's not related to ssh key. It suddenly stopped working

**Comment by Iceberry-qdd** [comment]
I also have this problem just now, It worked before.

![image](https://github.com/community/community/assets/59782644/e4af2e48-0bf9-45c0-ba0a-071aaa2edf36)

**Comment by cpzt** [comment]
@dizuncainiao I have the same promble. The temporary solution in Linux is as follows.
 ```bash
tee -a ~/.ssh/config <<EOF 
Host github.com
  Hostname ssh.github.com
  Port 443
EOF
```

**Comment by dizuncainiao** [comment, marked as answer]
这个问题解决了！

1. 在 _`C:\Users\{username}\.ssh`_ 目录下创建 _`config`_ 文件。注意：**没有后缀名**
2. 将以下内容复制到 _`config`_ 

    ```
    Host github.com
    User git
    Hostname ssh.github.com
    PreferredAuthentications publickey
    IdentityFile ~/.ssh/id_rsa
    Port 443
    ```
3. 在 Git Bash 中使用 `ssh -T git@github.com` 测试，不出意外的话，你可以看到以下的输出内容

    ![image](https://github.com/community/community/assets/40376068/4333d9e2-dfe5-41d1-a1b7-54bb75fd4882)
